examples/e-commerce/src/app/app.component.html

```diff
@@ -1,119 +1,187 @@
+        >
+        </ais-hits-per-page>
+      </header>
 
-                <div class="product-desc-wrapper">
-                  <div class="product-name">
-                    <ais-highlight attribute="name" [hit]="hit">
-                    </ais-highlight>
-                  </div>
+      <ais-hits>
+        <ng-template let-hits="hits">
+          <ol class="ais-Hits-list">
+            <li class="ais-Hits-item" *ngFor="let item of hits">
+              <article class="hit">
+                <header class="hit-image-container">
+                  <img [src]="item.image" [alt]="item.name" class="hit-image" />
+                </header>
 
-                  <div class="product-type">
-                    <ais-highlight attribute="type" [hit]="hit">
-                    </ais-highlight>
-                  </div>
+                <p class="hit-category">{{ item.categories[0] }}</p>
+                <h1>
+                  <ais-highlight attribute="name" [hit]="item"></ais-highlight>
+                </h1>
+                <p class="hit-description">
+                  <app-snippet
+                    attribute="description"
+                    [hit]="item"
+                  ></app-snippet>
+                </p>
 
-                  <div class="product-price">${{ hit.price }}</div>
-                  <div class="product-rating">
-                    <i
-                      *ngFor="let star of hit.stars"
-                      [ngClass]="{
-                        fa: true,
-                        'fa-star': star,
-                        'fa-star-o': !star
-                      }"
-                    >
-                    </i>
-                  </div>
-                </div>
+                <footer>
+                  <p>
+                    <span class="hit-em">$&nbsp;</span>
+                    <strong>{{ item.price }}&nbsp;</strong>
+                    <span class="hit-em hit-rating">
+                      <svg
+                        xmlns="http://www.w3.org/2000/svg"
+                        width="8"
+                        height="8"
+                        viewBox="0 0 16 16"
+                      >
+                        <path
+                          fill="#e2a400"
+                          fill-rule="evenodd"
+                          d="M10.472 5.008L16 5.816l-4 3.896.944 5.504L8 12.616l-4.944 2.6L4 9.712 0 5.816l5.528-.808L8 0z"
+                        ></path>
+                      </svg>
+                      {{ item.rating }}
+                    </span>
+                  </p>
+                </footer>
               </article>
-            </ng-template>
-          </ais-hits>
-        </main>
+            </li>
+          </ol>
+        </ng-template>
+      </ais-hits>
 
-        <ais-pagination></ais-pagination>
-      </div>
-    </div>
-  </ais-instantsearch>
-</div>
+      <footer class="container-footer">
+        <ais-pagination [padding]="2" [showFirst]="false"></ais-pagination>
```

Comment:
If we use `showFirst`, why not also use `showLast`?

Reply:
It was because of this:
```js
  @Input() public showFirst: boolean = true;
  @Input() public showLast: boolean = false;
```

`[showLast]="false"` is not needed here, but do you think we should put it there explicitly?

Reply:
I wasn't aware that `showLast` is `false` by default in Angular InstantSearch:

https://github.com/algolia/angular-instantsearch/blob/106df9ef102e7f701fcc3409a75ebc586b6bbf8f/src/pagination/pagination.ts#L109

This should be changed because the other flavors default value is `true`. We should fix that before going stable. Therefore, we can explictly specify `showLast` to `false` in the example to avoid having to change it after the release.

Reply:
fcdb4bb

refer to https://github.com/algolia/angular-instantsearch/pull/623